common/utils.cpp

```diff
@@ -121,6 +121,11 @@ bool guessResolution(const char* filename, int& w, int& h)
             sscanf(tokStart, "%d", &h);
         }
     }
+    //"w" or "h" overflows the int, that is dealt with as a bug.
+    if (w < 0 || h < 0) {
```

Comment:
What happens when you `sscanf` values in the range `long(std::numeric_limits<int>::max()) * 2 + 3`.. `long(std::numeric_limits<int>::max()) * 3 + 2` into an `int`?  This might be worth writing another unittest for.

Perhaps it's better to rewrite this function to use `std::strtol`, validate `errno` afterwards, then verify the result is in the range `0`..`std::numeric_limits<int>::max()` before assigning to w,h.